test/functional/gcp_shared_functions.go

```diff
@@ -0,0 +1,139 @@
+package functional
+
+//some functions below were taken from CRO
+import (
+	"context"
+	"fmt"
+	crov1 "github.com/integr8ly/cloud-resource-operator/apis/integreatly/v1alpha1"
+	croTypes "github.com/integr8ly/cloud-resource-operator/apis/integreatly/v1alpha1/types"
+	croResources "github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/constants"
+	"github.com/integr8ly/integreatly-operator/test/common"
+	errorUtil "github.com/pkg/errors"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net"
+	"net/url"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+const (
+	defaultNumberOfExpectedSubnets = 2
+	defaultIpRangeCIDRMask         = 22
+	managedLabelKey                = "red-hat-managed"
+	managedLabelValue              = "true"
+)
+
+func GetPostgresSqlInstancesIDsListFromCR(ctx context.Context, client client.Client, rhmi *integreatlyv1alpha1.RHMI) ([]string, []string) {
+	var foundErrors []string
+	var foundResourceIDs []string
+
+	expectedPSql := getExpectedPSql(rhmi.Spec.Type, rhmi.Name)
+
+	for _, r := range expectedPSql {
+		// get pSql cr
+		postgres := &crov1.Postgres{}
+		if err := client.Get(ctx, types.NamespacedName{Namespace: common.RHOAMOperatorNamespace, Name: r}, postgres); err != nil {
+			foundErrors = append(foundErrors, fmt.Sprintf("\nfailed to find %s postgres cr : %v", r, err))
+		}
+		// ensure phase is completed
+		if postgres.Status.Phase != croTypes.PhaseComplete {
+			foundErrors = append(foundErrors, fmt.Sprintf("\nfound %s postgres not ready with phase: %s, message: %s", r, postgres.Status.Phase, postgres.Status.Message))
+		}
+		// return resource id
+		resourceID, err := getCROAnnotation(postgres)
+		if err != nil {
+			foundErrors = append(foundErrors, fmt.Sprintf("\n%s postgres cr does not contain a resource id annotation: %v", r, err))
+		}
+		// populate the array
+		foundResourceIDs = append(foundResourceIDs, resourceID)
+	}
+	return foundResourceIDs, foundErrors
+}
+
+func getDefaultProjectID(ctx context.Context, c client.Client) (string, error) {
+	defaultProject, err := croResources.GetGCPProject(ctx, c)
+	if err != nil {
+		return "", errorUtil.Wrap(err, "failed to retrieve project from cluster")
+	}
+	if defaultProject == "" {
+		return "", errorUtil.New("failed to retrieve project from cluster, project ID is not defined")
+	}
+	return defaultProject, nil
+}
```

Comment:
I'm unsure if this wrapper function is needed - currently the `croResources.GetGCPProject(ctx, c)` function will either return an `error` or a `projectID` so I'm not sure what this function adds over calling the `croResources` version directly.

Reply:
Done. Thank you